analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantModifier.cs

```diff
@@ -206,75 +200,38 @@ private static void MarkAllUnsafeBlockInside(SyntaxNode container, SyntaxNodeAna
+            Modifiers(memberDeclaration).FirstOrDefault(x => x.IsKind(SyntaxKind.UnsafeKeyword));
 
         private static void CheckTypeDeclarationForRedundantPartial(SyntaxNodeAnalysisContext context)
         {
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
-
-            if (!typeDeclaration.Modifiers.Any(SyntaxKind.PartialKeyword)
-                || context.ContainingSymbol.Kind != SymbolKind.NamedType)
-            {
-                return;
-            }
-
-            if (context.SemanticModel.GetDeclaredSymbol(typeDeclaration) is { } typeSymbol
+            if (typeDeclaration.Modifiers.Any(SyntaxKind.PartialKeyword)
+                && context.ContainingSymbol.Kind == SymbolKind.NamedType
+                && context.SemanticModel.GetDeclaredSymbol(typeDeclaration) is { } typeSymbol
                 && typeSymbol.DeclaringSyntaxReferences.Length <= 1)
             {
                 var keyword = typeDeclaration.Modifiers.First(m => m.IsKind(SyntaxKind.PartialKeyword));
                 context.ReportIssue(Diagnostic.Create(Rule, keyword.GetLocation(), "partial", "gratuitous"));
             }
         }
 
-        private static SyntaxTokenList GetModifiers(MemberDeclarationSyntax memberDeclaration) =>
+        private static SyntaxTokenList Modifiers(MemberDeclarationSyntax memberDeclaration) =>
             memberDeclaration switch
             {
-                MethodDeclarationSyntax methodDeclaration => methodDeclaration.Modifiers,
-                PropertyDeclarationSyntax propertyDeclaration => propertyDeclaration.Modifiers,
-                IndexerDeclarationSyntax indexerDeclaration => indexerDeclaration.Modifiers,
-                EventDeclarationSyntax eventDeclaration => eventDeclaration.Modifiers,
-                EventFieldDeclarationSyntax eventFieldDeclaration => eventFieldDeclaration.Modifiers,
+                BasePropertyDeclarationSyntax propertyDeclaration => propertyDeclaration.Modifiers,
+                BaseMethodDeclarationSyntax methodDeclaration => methodDeclaration.Modifiers,
+                BaseFieldDeclarationSyntax fieldDeclaration => fieldDeclaration.Modifiers,
+                DelegateDeclarationSyntax delegateDeclaration => delegateDeclaration.Modifiers,
+                TypeDeclarationSyntax typeDeclaration => typeDeclaration.Modifiers,
                 _ => default
             };
 
         private static void CheckSealedMemberInSealedClass(SyntaxNodeAnalysisContext context)
         {
-            var memberDeclaration = (MemberDeclarationSyntax)context.Node;
-            if (GetModifiers(memberDeclaration) is var modifiers
+            if (Modifiers((MemberDeclarationSyntax)context.Node) is var modifiers
                 && modifiers.Any(SyntaxKind.SealedKeyword)
-                && context.ContainingSymbol is {IsSealed: true, ContainingType: {IsSealed: true}})
+                && context.ContainingSymbol.ContainingType is { IsSealed: true })
```

Comment:
Just to make sure this change is intended:
**Before**
* ContainingSymbol.IsSealed: true
* ContainingSymbol.ContainingType.IsSealed: true

**After**

* ContainingSymbol.ContainingType.IsSealed: true

Reply:
Yes it was intended to bump coverage, because if the containing type is sealed, then every member symbol is marked as sealed too => we don't need to re-check that.